--- transpose.toy-dialect.mlir (no -opt)
+++ transpose.toy-dialect.opt.mlir (-opt)
@@ -1,14 +1,7 @@
 module {
-  toy.func private @transpose_transpose(%arg0: tensor<*xf64>) -> tensor<*xf64> {
-    %0 = toy.transpose(%arg0 : tensor<*xf64>) to tensor<*xf64>
-    %1 = toy.transpose(%0 : tensor<*xf64>) to tensor<*xf64>
-    toy.return %1 : tensor<*xf64>
-  }
   toy.func @main() {
     %0 = toy.constant dense<[[1.000000e+00, 2.000000e+00, 3.000000e+00], [4.000000e+00, 5.000000e+00, 6.000000e+00]]> : tensor<2x3xf64>
-    %1 = toy.reshape(%0 : tensor<2x3xf64>) to tensor<2x3xf64>
-    %2 = toy.generic_call @transpose_transpose(%1) : (tensor<2x3xf64>) -> tensor<*xf64>
-    toy.print %2 : tensor<*xf64>
+    toy.print %0 : tensor<2x3xf64>
     toy.return
   }
 }
